### DISCUSSION TITLE
Does this work with cloudflare?

### DISCUSSION THREAD

**Discussion by janat08** [logto-io/logto#1675] [Q&A]
Im looking to use DO as sessions, but I'd use the logto package if it works.

**Comment by gao-sun** [comment]
we haven't tested yet. you can try it and let us know if anything is not working.

**Comment by kinthaiofficial** [comment]
Logto works well with Cloudflare in a few different deployment patterns:

**1. Logto behind Cloudflare proxy** (most common):
```
User → Cloudflare CDN → Your origin (Logto)
```

Key considerations:
- Set Cloudflare SSL/TLS mode to "Full (strict)" — Logto requires HTTPS
- Add Logto's domain to Cloudflare, point to your server IP
- Cloudflare handles DDoS protection and CDN for static assets

**2. Cloudflare Pages + Logto Cloud**:
```yaml
# If using Logto Cloud, just configure CORS in Logto Console
# Add your Cloudflare Pages domain as an allowed origin
```

**3. Workers AI integration** (for serverless + auth):
```typescript
// Cloudflare Worker with Logto JWT verification
import { createRemoteJWKSet, jwtVerify } from 'jose';

const JWKS = createRemoteJWKSet(
  new URL('https://your-logto-endpoint.com/oidc/jwks')
);

export default {
  async fetch(request: Request): Promise<Response> {
    const token = request.headers.get('Authorization')?.replace('Bearer ', '');
    
    if (!token) return new Response('Unauthorized', { status: 401 });
    
    const { payload } = await jwtVerify(token, JWKS, {
      issuer: 'https://your-logto-endpoint.com/oidc',
      audience: 'your-api-resource',
    });
    
    // payload.sub = user ID, proceed with request
    return handleRequest(request, payload);
  }
};
```

**Common Cloudflare + Logto gotcha**: Cloudflare's Rocket Loader can interfere with Logto's SDK initialization. Disable it for pages with Logto sign-in:
```html
<script data-cfasync="false" src="logto-sdk.js"></script>
```

What's your specific setup — self-hosted Logto or Logto Cloud?
